### PR TITLE
Add escape false to PaginationHelper

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves https://github.com/cakephp/cakephp/issues/9310

First draft of yesterday evening.
If the idea is accepted I can add some tests later on.

### CODE REVIEW

File: src/View/Helper/PaginatorHelper.php [hunk 1]
```diff
@@ -468,16 +468,29 @@ public function sort($key, $title = null, array $options = [])
...
      * @link http://book.cakephp.org/3.0/en/views/helpers/paginator.html#generating-pagination-urls
      */
     public function generateUrl(array $options = [], $model = null, $full = false)
     {
         $paging = $this->params($model);
         $paging += ['page' => null, 'sort' => null, 'direction' => null, 'limit' => null];
+
+        $defaults = [
+            'escape' => true,
+            'fullBase' => $full
```
Comment (markstory) [inline src/View/Helper/PaginatorHelper.php]: This means you can never have URL parameters with these names.

Comment (dereuromark) [inline src/View/Helper/PaginatorHelper.php]: So we need to use a separate $config here then for the url options only.

**Review by markstory — CHANGES_REQUESTED** [review]
Looks good, you need some tests though.

**Comment by dereuromark** [comment]
Tests added.

**Comment by lorenzo** [comment]
Good job!

**Pull request merged**